### PR TITLE
feat: add requirement_test workflow to template

### PR DESCRIPTION
**Pull request by nandinivij**
Please review -

Added requirement test unit tests to the common template. these currently run in many addons for eg Juniper
https://github.com/splunk/splunk-add-on-for-juniper/pull/277

### CODE REVIEW

File: enforce/.github/workflows/requirements_unit_test.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Requirement unit tests workflow
+on: push
+jobs:
+  build:
+    name: Run unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
```
Comment (arys-splunk) [inline enforce/.github/workflows/requirements_unit_test.yml:8]:
```suggestion
      - uses: actions/checkout@v2
```

File: enforce/.github/workflows/requirements_unit_test.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Install Python 3
+        uses: actions/setup-python@v1
```
Comment (arys-splunk) [inline enforce/.github/workflows/requirements_unit_test.yml:10]:
```suggestion
        uses: actions/setup-python@v2
```

File: enforce/.github/workflows/requirements_unit_test.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        with:
+            python-version: 3.8
```
Comment (arys-splunk) [inline enforce/.github/workflows/requirements_unit_test.yml:12]: Any specific reason to run it 3.8?

File: enforce/.github/workflows/requirements_unit_test.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      - name: Check file existence
+        id: check_files
+        uses: andstor/file-existence-action@v1
```
Comment (arys-splunk) [inline enforce/.github/workflows/requirements_unit_test.yml:15]: Can we check it without using external dependency?

File: enforce/.github/workflows/requirements_unit_test.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        with:
+          files: "tests/requirement_test"
+      - id: requirement_test_pre_requisite_test
+        if: steps.check_files.outputs.files_exists == 'true'
+        uses:  splunk/requirement-files-unit-tests@main
```
Comment (arys-splunk) [inline enforce/.github/workflows/requirements_unit_test.yml:20]: I think it will be better to use version instead of running code from main branch. WDYT?

**Review by arys-splunk — CHANGES_REQUESTED** [review]

File: enforce/.github/workflows/requirements_unit_test.yml (hunk 3 above)
Comment (nandinivij) [reply enforce/.github/workflows/requirements_unit_test.yml:12]: we need a particular version set used what was latest at that time

Comment (arys-splunk) [reply enforce/.github/workflows/requirements_unit_test.yml:12]: We have Python 3.7 everywhere, can we use it here as well?

**Review by arys-splunk — COMMENTED** [review]
I am not sure that we need to put requirement tests linter as a separate workflow, not as a part of build-test-release. @uoboda-splunk WDYT?

https://github.com/splunk/github-workflow-build-test-release/blob/main/.github/workflows/reusable-build-test-release.yml

**Comment by nandinivij** [comment]
> I am not sure that we need to put requirement tests linter as a separate workflow, not as a part of build-test-release. @uoboda-splunk WDYT?
> 
> https://github.com/splunk/github-workflow-build-test-release/blob/main/.github/workflows/reusable-build-test-release.yml

@arys-splunk  These tests are supposed to run when logs are added to the repo without creating a PR, which is not true with the build test release.

**Comment by arys-splunk** [comment]
> > I am not sure that we need to put requirement tests linter as a separate workflow, not as a part of build-test-release. @uoboda-splunk WDYT?
> > https://github.com/splunk/github-workflow-build-test-release/blob/main/.github/workflows/reusable-build-test-release.yml
> 
> @arys-splunk These tests are supposed to run when logs are added to the repo without creating a PR, which is not true with the build test release.

Shouldn't they run only when we need to run requirement tests?

And if I have requirement tests in the repo, we are not going to run them? I think we should, but as a prerequisite to the requirement tests themselves.

**Comment by nandinivij** [comment]
> > > I am not sure that we need to put requirement tests linter as a separate workflow, not as a part of build-test-release. @uoboda-splunk WDYT?
> > > https://github.com/splunk/github-workflow-build-test-release/blob/main/.github/workflows/reusable-build-test-release.yml
> > 
> > 
> > @arys-splunk These tests are supposed to run when logs are added to the repo without creating a PR, which is not true with the build test release.
> 
> Shouldn't they run only when we need to run requirement tests?
> 
> And if I have requirement tests in the repo, we are not going to run them? I think we should, but as a prerequisite to the requirement tests themselves.

They are pre-requisite to the requirement test if these tests are not passing for eg: the XML formatting fails we should not run the requirement test.  We want it to run as soon as logs are added to the repo. This is the ask for these unit tests

**Comment by arys-splunk** [comment]
That makes sense, but it does not make this linter workflows unique enough to be in the separate file and it can be easily integrated with build-test-release, which will give us less maintenance in the future. And this linter will run in the same time as, for example, `quality-lint`.

**Comment by arys-splunk** [comment]
And I think that if developers need fast feedback about their samples, they can and should run all the checks locally. So, I think we need to make it super easy to run locally as well.

**Review by arys-splunk — CHANGES_REQUESTED** [review]
@nandinivij Can you please change the name of the repository (https://github.com/splunk/requirement-files-unit-tests) to be prefixed with "addonfactory"?That's the convention we have for the repositories that support add-ons. This will probably break all existing integrations, so we will need to do this before merging.

**Comment by nandinivij** [comment]
> @nandinivij Can you please change the name of the repository (https://github.com/splunk/requirement-files-unit-tests) to be prefixed with "addonfactory"?That's the convention we have for the repositories that support add-ons. This will probably break all existing integrations, so we will need to do this before merging.

Hi @arys-splunk we cannot change the name. This repo is public we will have to go through a process with oss committee to change the name which we want to avoid. cc @drnkllr

**Comment by nandinivij** [comment]
> And I think that if developers need fast feedback about their samples, they can and should run all the checks locally. So, I think we need to make it super easy to run locally as well.

Hi @arys-splunk Centralized tests were not designed to run locally. We needed these tests when we migrated logs from logs repo to TA to run on CI. If we need them to run locally on each TA that will be a new requirement and task. 

Requirement test workflow existed before other workflows were added with the move from .circleci to Github recently and are built by the GDI QA team. I discussed moving this workflow with other workflows with @drnkllr. It was advised we should run them earlier.

File: enforce/.github/workflows/requirements_unit_test.yml (hunk 5 above)
Comment (uoboda-splunk) [reply enforce/.github/workflows/requirements_unit_test.yml]: I think it should be a part of `build-test-release` workflow, WDYT?

**Comment by uoboda-splunk** [comment]
> > > > I am not sure that we need to put requirement tests linter as a separate workflow, not as a part of build-test-release. @uoboda-splunk WDYT?
> > > > https://github.com/splunk/github-workflow-build-test-release/blob/main/.github/workflows/reusable-build-test-release.yml
> > > 
> > > 
> > > @arys-splunk These tests are supposed to run when logs are added to the repo without creating a PR, which is not true with the build test release.
> > 
> > 
> > Shouldn't they run only when we need to run requirement tests?
> > And if I have requirement tests in the repo, we are not going to run them? I think we should, but as a prerequisite to the requirement tests themselves.
> 
> They are pre-requisite to the requirement test if these tests are not passing for eg: the XML formatting fails we should not run the requirement test. We want it to run as soon as logs are added to the repo. This is the ask for these unit tests, it is required that dev get to know linter failures as soon as possible.

Yes I understand the idea of running it in separate workflow and some benefits from it. However we have all similar jobs in `build-test-release` workflow and it would be strange to have an exception for this. As @arys-splunk mentioned I also think this job should be pre-requisite to the requirement tests in `build-test-release`. @rfaircloth-splunk what do you think?

**Comment by nandinivij** [comment]
As decided we are moving the workflow to build test release. We will need a task to delete the current workflow from all the TA's

**Pull request closed without merging**